lib/std/array_list.zig

```diff
@@ -1,4 +1,4 @@
-const std = @import("std.zig");
+const std = @import("std");
```

Comment:
I don't like this change because it makes the standard library special and different from normal user libraries which will not have themselves mapped into their own import table like this. The reason std lib has itself mapped is for this check to work: https://github.com/ziglang/zig/blob/67d684d89aeb0bb3cfa86c57e8d77359d45743fa/lib/std/os.zig#L36-L38

Reply:
Somewhere `@import("std")` is used, and somewhere `@import("std.zig")`.
It's confused.

Reply:
It should be `@import("std.zig")` everywhere except for that one spot mentioned above